### ISSUE TITLE
v1.1.1 fails to generate requirements from poetry lockfile

### ISSUE THREAD

**Issue by macarr** [thoth-station/micropipenv#193]
**Describe the bug**

micropipenv fails to generate requirements using poetry method for certain configurations in 1.1.1

**To Reproduce**

given:
```
[tool.poetry]
name = "test"
version = "0.1.0"
description = "test"
authors = ["test"]

[tool.poetry.dependencies]
python = "3.8.10"
flask-restx = "^0.4"
webargs = "^7.0"
gunicorn = "^19.9"
Flask-PyMongo = "^2.3.0"
py-healthcheck = "^1.2"
invoke = "^1.6"
pyyaml = "^5.4"

[tool.poetry.dev-dependencies]
pytest = "*"
pytest-mock = "*"
pytest-flask = "*"
requests = "*"
requests-mock = "*"
responses = "*"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

when I generate the poetry lockfile using `poetry lock` on poetry version 1.1.10

when I execute `micropipenv requirements --method poetry --no-dev`

I get
`'atomicwrites'`
and then exit code 3

when I execute `micropipenv requirements --method poetry --no-default`

I get
`local variable 'category' referenced before assignment`

**Expected behavior**

On micropipenv 1.1.0, the requirements files are generated as expected

**Comment by frenzymadness** [comment]
Thanks for the report. I'm gonna investigate it.

**Comment by frenzymadness** [comment]
Could you please test my fix in #195?
